### ISSUE TITLE
Externalize editor localization text

### ISSUE THREAD

**Issue by subsoap** [defold/defold#5453]
**Is your feature request related to a problem? Please describe (REQUIRED):**
Currently interface text is baked into code. It makes adding translations of the editor to not be possible.

**Describe the solution you'd like (REQUIRED):**
Add a key system for grabbing localization strings based on locale. Default to English, have multiple places to toggle the language such as on the launcher / settings. Either allow hotswap on text / reload UI / or force editor restart to update language.

Should also support extra languages / language overrides within the .defold user folder. This is useful for people donating translations to easily see updates. The "Update CSS" button could be changed to mean everything in the .defold user folder.

All files related to language should be under a locale folder such as en-us with a folder above it have for example a language list file along with their short and long names along with which should be default.

**Describe alternatives you've considered (REQUIRED):**
People could fork Defold and change the English text to other languages. This should be avoided, it should be done in a clean and centralized way.

**Additional context (OPTIONAL):**
Some languages such as RTL would require extra care.

This should have more discussion about implementation / design before anyone starts on it. There needs to be proposals on exact implementation people can agree on since it needs to be as good as possible of a process for translators to contribute to.

**Comment by subsoap** [comment]
I feel that text key strings should be written in English still.

```
:title "Unable to Open Resource"
```
-> 
```
:title System.getLocale("unable_to_open_resource")
```
I don't know enough about Clojure to know if this would work but something generally like this.

Potentially include the area the text string is located to help with organization

```
:title System.getLocale("app_view_unable_to_open_resource")
```

One trick we do to find text that is not setup right is make all calls to get localized strings to return something like ZZZZZ then text that still shows up in English obviously still needs to be moved over.

However the CSS updates should be used to update text too if that is possible. I don't know yet on that.

**Comment by subsoap** [comment]
For key names, using dashes instead of underscores to separate words

```
"app-view-unable-to-open-resource"
```

instead of

```
"app_view_unable_to_open_resource"
```

seems would work better to fit existing style.

**Comment by subsoap** [comment]
Seems like it could look something like this for getting. It seems a little annoying, but there is already loads of code with a similar pattern. 

```
 [editor.locale :as locale]
```
```
 (locale/get-locale locale "app-view-unable-to-open-resource")
```

Like I'd prefer just get and not get-locale but get-locale is closer to existing pattern.

Ref with prefs

https://github.com/defold/defold/blob/dev/editor/src/clj/editor/prefs.clj

**Comment by zuiaiyangyang3** [comment]
Yes, the editor interface cannot be understood and cannot experience this engine
Can the document be later in Chinese? After all, the browser can translate it
But if I can't understand the editor interface, I really can't understand it

**Comment by AGulev** [comment]
related https://github.com/defold/defold/issues/10864

**Comment by britzl** [comment]
https://defold.com/manuals/editor/#editor-language

**Comment by vlaaad** [comment]
I'm working on this issue right now (localizing error messages)

**Comment by britzl** [comment]
> I'm working on this issue right now (localizing error messages)

Don't we have a separate task for this? I'd like to avoid having unclear large open tasks which we could break down into multiple smaller tasks.
